src/sqs/index.ts

```diff
@@ -268,21 +268,25 @@ export class SqsQueue<MessagePayload> {
 			WaitTimeSeconds: this.longPollingIntervalSec
 		};
 
-		//Get messages from the queue with long polling enabled
-		await this.sqs.receiveMessage(params)
-			.promise()
-			.then(async result => {
-				await this.handleSqsResponse(result, listenerContext)
-			})
-			.catch((err) => {
-				listenerContext.log.error({err}, `Error receiving message from SQS queue, queueName ${this.queueName}`);
-			})
-			.finally( () =>
-				this.listen(listenerContext)
-			);
+		try {
+			//Get messages from the queue with long polling enabled
+			const result = await this.sqs.receiveMessage(params)
+				.promise();
+
+			await this.handleSqsResponse(result, listenerContext)
+		} catch(err) {
+			listenerContext.log.error({err}, `Error receiving message from SQS queue, queueName ${this.queueName}`);
+			//In case of aws client error we wait for the long polling interval to prevent bombarding the queue with failing requests
+			await new Promise(resolve => setTimeout(resolve, this.longPollingIntervalSec * 1000));
```

Comment:
I had to change it here, because I found when testing that if AWS Client for some reason fails (Service Unavailable due to DNS error, for example) 

We will retry calling the queue straightaway and will spam logs with errors

Reply:
I've added a delay instead of calling listener in `setTimeout` function, because I wanted to keep it in the `finally` block